ls1intum/Artemis
# Add build logs to database.

### Checklist

### Motivation and Context
See #2067:
Currently, Artemis has to fetch the build logs from the CI server (like Bamboo) every time the client requests these logs.
This is not optimal as on the one hand, it causes additional requests to the CI server. On the other hand, the build logs might be cleaned on the CI server and are thus not available anymore.

### Description
We store the build logs of failed builds within Artemis. If the build plans are cleaned on the CI server, they are still persisted in Artemis.
We do not have to store logs of builds that did not fail as the feedback is then displayed directly within Artemis.


Artemis will still include the current behavior (fetching the build logs) as existing submissions do not have build logs stored in the database and we currently only plan to implement this behavior for Bamboo, not for Jenkins.


The database changes have been approved in #2067.

### Steps for Testing
1. Participate in a programming exercise using the online code editor
2. Produce a failed build -> The build log should be shown & should be correctly formatted
3. If testing locally (requires setup for the programming exercises): Ensure that the build logs are persisted in the database

### Test Coverage
* BambooService.java: 33% (previously 32%)
